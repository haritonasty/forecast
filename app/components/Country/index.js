/**
 *
 * Country
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { colorTheme, fontSize, fontWeight } from '../../utils/style';

export const StyledCountry = styled.button`
  padding: 3px 5px;
  font-family: inherit;
  font-weight: ${fontWeight.default};
  font-size: ${fontSize.medium}px;
  border-bottom: 2px solid
    ${p => (p.open ? colorTheme.base : colorTheme.primary)};
  margin: 4px 0;
  color: ${colorTheme.primary};
  cursor: pointer;
  outline-color: ${colorTheme.primary};
`;

function Country({ value, children, toggleItem, isOpen }) {
  return (
    <>
      <StyledCountry
        onClick={toggleItem}
        open={isOpen}
        title={isOpen ? 'Hide cities' : 'Show cities'}
      >
        {value}
      </StyledCountry>
      {isOpen && children}
    </>
  );
}

Country.propTypes = {
  value: PropTypes.string.isRequired,
};

export default Country;
