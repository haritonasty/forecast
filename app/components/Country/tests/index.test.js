import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Country, { StyledCountry } from '../index';

describe('<Country />', () => {
  it('should render StyledCountry', () => {
    const props = { value: 'Russia', isOpen: true };
    const renderedComponent = shallow(<Country {...props} />);
    expect(renderedComponent.contains(<StyledCountry />)).toBeDefined();
  });

  it('should render children', () => {
    const props = {
      value: 'Russia',
      isOpen: true,
      children: ['Russia'],
    };
    const renderedComponent = renderer.create(<Country {...props} />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should not render children', () => {
    const props = {
      value: 'Russia',
      isOpen: false,
      children: ['Moscow'],
    };
    const renderedComponent = renderer.create(<Country {...props} />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
