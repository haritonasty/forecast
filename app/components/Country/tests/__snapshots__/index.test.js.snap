// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Country /> should not render children 1`] = `
<button
  className="Country__StyledCountry-tfl0yw-0 jAVkNj"
  open={false}
  title="Show cities"
>
  Russia
</button>
`;

exports[`<Country /> should render children 1`] = `
Array [
  <button
    className="Country__StyledCountry-tfl0yw-0 fjFMzj"
    open={true}
    title="Hide cities"
  >
    Russia
  </button>,
  "Russia",
]
`;
